pmlc/conversion/tile_to_pxa/tile_to_pxa.cc

```diff
@@ -727,6 +729,79 @@ struct IndexOpConversion : public OpConversionPattern<IndexOp> {
   }
 };
 
+struct ReshapeOpConversion : public OpConversionPattern<ReshapeOp> {
+  using OpConversionPattern<ReshapeOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(ReshapeOp op, llvm::ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const override {
+    IVLOG(2, "ReshapeOpConversion::matchAndRewrite>");
+
+    ModuleOp module(op.getParentOfType<ModuleOp>());
+
+    // Create an adaptor, to interpret the operands
+    ReshapeOpOperandAdaptor adaptor(operands);
+
+    auto loc = op.getLoc();
+    TypeConverter typeConverter;
+    auto resultType =
+        typeConverter.convertType(op.result().getType()).cast<MemRefType>();
+
+    // Make an allocation for the output
+    auto resultMemRef = rewriter.create<AllocOp>(loc, resultType).getResult();
+
+    // Get an unranked memref type for the same element type.
+    // Crude hack for getting it going: assume f32 all the time
+    auto elementType = rewriter.getF32Type();
+    auto unrankedType = UnrankedMemRefType::get(elementType, /*memorySpace*/ 0);
+
+    // The source is adaptor.tensor()
+    auto source =
+        rewriter.create<MemRefCastOp>(loc, adaptor.tensor(), unrankedType);
+    // The dest is op.result()
+    auto dest = rewriter.create<MemRefCastOp>(loc, op.result(), unrankedType);
+
+    // The new shape is adaptor.dims()
+    auto i32Type = rewriter.getIntegerType(32);
+    Value count = adaptor.dims()[0];
```

Comment:
Rank 0 tensors do exist.  If someone does a reshape on one, this will probably crash.  Probably better to make a constant value of 1 and then multiply by every dim.